trunk/modules/python/dataset/ingest/labeled.py

```diff
@@ -140,7 +140,13 @@ def hdf5Dataset(filepath, holdoutPercentage=.05, minTest=5,
     # Here we create the handles to the data buffers. This operates on the
     # assumption the dataset may not fit entirely in memory. The handles allow
     # data to overflow and ultimately be stored entirely on disk. 
-    imageShape = list(getImageDims(train[0][0], log))
+
+    # find the most common file size and use to get the image
+    # shape
+    commonSize = mostCommon(train, lambda t: os.path.getsize(t[0]))
+    exampleImage = (t for t in train
+                    if os.path.getsize(t[0]) == commonSize).next()

```

Comment:
Also use filter() here. Also I'm getting an error:

AttributeError: 'generator' object has no attribute 'next'

Please look at this. If there is already a hdf5 created, this section of code with not be executed, so try changing the --batch or --holdout command line arguments to always run this section of the code.